src-tauri/src/backend/db_ops.rs

```diff
@@ -250,21 +250,44 @@ impl DBOps {
         Ok(count as u64)
     }
 
-    /// Removes the specified hash from the `signatures` table.
+    /// Removes a vec of specified hashes from the `signatures` table.
     ///
     /// # Examples
     ///
     /// ```
     /// use rusqlite::Connection;
     /// use virus_scanner::backend::db_ops::DBOps;
     /// let db_ops = DBOps::new("signatures.db").unwrap();
-    /// assert!(db_ops._remove_hash("abcd1234").is_ok());
+    /// assert!(db_ops._remove_hash(vec!["abcd1234".to_owned()]).is_ok());
     /// ```
-    pub fn _remove_hash(&self, hash_str: &str) -> Result<(), rusqlite::Error> {
-        self.db_conn.execute(
-            &format!("DELETE FROM {} WHERE hash = ?", crate::DB_TABLE),
-            [hash_str],
-        )?;
+    pub fn remove_hashes(&mut self, hashes: &Vec<String>) -> Result<(), rusqlite::Error> {
+        self.init_table()?;
+        info!("Removing {} hashes...", hashes.len());
+        let transact = self.db_conn.transaction()?;
+
+        let big_tic = time::Instant::now();
+        let mut removed = 0;
+        let mut skipped = 0;
+        for hash in hashes {
+            match transact.execute(
+                &format!("DELETE FROM {} WHERE hash = ?", crate::DB_TABLE),
+                [hash.clone()],
+            ) {
+                Ok(_) => removed += 1,
+                Err(err) => {
+                    warn!("Got {err} when trying to insert {hash}. Skipping...");
```

Comment:
Probably should be "When trying to delete hash" right?

Reply:
oh yeah youre right

Reply:
fixed with 8f300aab5fa080d2f97879bdda9257d592cd96a0